x-pack/filebeat/input/awss3/input.go

```diff
@@ -410,5 +413,35 @@ func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
 	}
 }
 
+func pollSqsUtilizationMetric(ctx context.Context, receiver *sqsReader, metricReporterChan chan int64) {
+	defer close(metricReporterChan)
+
+	t := time.NewTicker(5 * time.Second)
+	defer t.Stop()
+
+	var utilizedNanos int64
+	go func() {
+		for elem := range metricReporterChan {
+			atomic.AddInt64(&utilizedNanos, elem)
+		}
+	}()
+
+	lastTick := time.Now()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case tick := <-t.C:
+			denom := float64(tick.Sub(lastTick)) * float64(receiver.maxMessagesInflight)
```

Comment:
How about naming this variable something more descriptive like `maxUtilization`.